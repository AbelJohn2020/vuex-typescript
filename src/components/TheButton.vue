<template>
  <button :type="typeButton" :disabled="isDisabled">{{ name }}</button>
</template>

<script>
export default {
    props: {
        typeButton: {
            type: String,
        },
        isDisabled: {
            type: Boolean,
        },
        name: {
            type: String,
            required: true,
        }
    }
}
</script>

<style>
button {
  font: inherit;
  border: 1px solid #0076bb;
  background-color: #0076bb;
  color: white;
  cursor: pointer;
  padding: 0.75rem 2rem;
  border-radius: 30px;
  text-transform: capitalize;
}

button:hover,
button:active {
  border-color: #002350;
  background-color: #002350;
}

button:disabled {
    border-color: #AAAAAA;
    color: #AAAAAA;
    background-color: white;
    cursor: default;
}
</style>