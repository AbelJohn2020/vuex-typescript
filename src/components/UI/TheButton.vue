<template>
  <button :type="typeButton" :disabled="isDisabled" :class="this.$store.state.language === 'english' ? 'english' : 'español'">{{ name }}</button>
</template>

<script>
export default {
    props: {
        typeButton: {
            type: String,
        },
        isDisabled: {
            type: Boolean,
        },
        name: {
            type: String,
            required: true,
        }
    }
}
</script>

<style>
button {
  font: inherit;
  font-weight: bold;
  cursor: pointer;
  padding: 0.44rem 1.2rem;
  border-radius: 8px;
  text-transform: capitalize;
  color: white;
}

button:disabled {
  color: white;
  cursor: default;
}

.english {
  border: 1px solid #0076bb;
  background-color: #0076bb;
}

.español {
  border: 1px solid #A51C30;
  background-color: #A51C30;
}

.english:hover,
.english:active {
  border-color: #002350;
  background-color: #002350;
}

.español:hover,
.español:active {
  border-color: #881C30;
  background-color: #881C30;
}

.english:disabled {
    border-color: rgba(0,149,246,.3);
    background-color: rgba(0,149,246,.3);
}

.español:disabled {
    border-color: rgba(149,32,42,.3);
    background-color: rgba(149,32,42,.3);
}

</style>