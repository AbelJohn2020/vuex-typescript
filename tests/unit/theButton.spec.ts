import { mount } from "@vue/test-utils";
import TheButton from '@/components/UI/TheButton.vue';

describe('Testing Button component', () => { 
    test('Should have a name', () => { 
        const wrapper = mount(TheButton, {
            props: {
                name: 'test button',
            }
        });

        expect(wrapper.html()).toContain('test button')
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Should have name and type', () => { 
        const wrapper = mount(TheButton, {
            props: {
                typeButton: '',
                name: 'test button',
            }
        });
    
        expect(wrapper.html()).toContain('test button');

        // si en caso no tiene un tipo entonces se quedara con el default.
        expect(wrapper.element.getAttribute('type')).toEqual('submit');
        expect(wrapper.html()).toMatchSnapshot();
     })

     test('Should have name, type and disabled', () => { 
        const wrapper = mount(TheButton, {
            props: {
                typeButton: '',
                isDisabled: true,
                name: 'test dissabled button',
            },
        });

        expect(wrapper.html()).toContain('test dissabled button');

        expect(wrapper.vm.typeButton).toBe('');
        expect(wrapper.element.getAttribute('type')).toBe('submit');

        expect(wrapper.vm.isDisabled).toEqual(true);
        expect(wrapper.html()).toMatchSnapshot();
      })
 })