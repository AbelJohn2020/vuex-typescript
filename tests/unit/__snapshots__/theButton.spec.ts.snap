// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Should have a name 1`] = `<button>test button</button>`;

exports[`Testing Button component Should have name and type 1`] = `<button type="submit">test button</button>`;

exports[`Testing Button component Should have name, type and disabled 1`] = `<button type="submit" disabled="">test dissabled button</button>`;
